### PR TITLE
Clone source from mirror for DartLab integration tests

### PR DESCRIPTION
**Pull request by JoeRobich**
There were concern about automatically running this pipeline and building from GitHub. I have updated the pipeline to pull source from our internal mirror instead.

Test run: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5632203&view=results (microsoft)

### CODE REVIEW

File: azure-pipelines-integration-dartlab.yml [hunk 1]
```diff
@@ -54,6 +62,7 @@ stages:
         filePath: $(DartLab.Path)\Scripts\VisualStudio\Build\Get-VisualStudioDropName.ps1
         arguments: -DropNamePrefix 'Products' -VstsDropUrlsJson '$(Pipeline.Workspace)\VisualStudioBuildUnderTest\BuildArtifacts\VstsDropUrls.json' -OutVariableName 'VisualStudio.BuildUnderTest.ProductsDropName'
     deployAndRunTestsStepList:
+    - checkout: RoslynMirror
```
Comment (JoeRobich) [inline azure-pipelines-integration-dartlab.yml:65]: Since we do not specify `- checkout: self` we are pulling the source from our mirror instead of GH.

**Comment by JoeRobich** [comment]
@jaredpar Please take a look

**Review by RikkiGibson — APPROVED** [review]
LGTM. Would be nice to use the checkout task (modify to parameterize what repo we use?) but not important.

File: azure-pipelines-integration-dartlab.yml (hunk 1 above)
Comment (jaredpar) [reply azure-pipelines-integration-dartlab.yml:65]: Does this run the full `git` checkout or the modified one that we built which does a minimal checkout?

Comment (JoeRobich) [reply azure-pipelines-integration-dartlab.yml:65]: This is AzDO running its full `git` checkout.  I will have to see how extra git resources are managed to determine if we can update our checkout task.

Comment (jaredpar) [reply azure-pipelines-integration-dartlab.yml:65]: I don't think it's anything that should block this change but is something that we should keep an eye on. The frequency of our checkouts in normal CI made this a problem, doubt this hits as frequently hence less of a trigger point.

**Pull request merged**